src/main.rs

```diff
@@ -58,51 +68,94 @@ impl BFInterp {
+                    self.int_bin_operations(&udiv.operand0, &udiv.operand1, &udiv.dest, BinOps::Div)
+                }
+                Instruction::SDiv(sdiv) => {
+                    self.int_bin_operations(&sdiv.operand0, &sdiv.operand1, &sdiv.dest, BinOps::Div)
+                }
+                Instruction::URem(urem) => {
+                    self.int_bin_operations(&urem.operand0, &urem.operand1, &urem.dest, BinOps::Rem)
+                }
+                Instruction::SRem(srem) => {
+                    self.int_bin_operations(&srem.operand0, &srem.operand1, &srem.dest, BinOps::Rem)
+                }
+                Instruction::FSub(fsub) => {
+                    self.fl_bin_operations(&fsub.operand0, &fsub.operand1, &fsub.dest, BinOps::Sub)
+                }
+                Instruction::SExt(sext) => self.sext(&sext.operand, &sext.to_type, &sext.dest),
+                Instruction::FMul(fmul) => {
+                    self.fl_bin_operations(&fmul.operand0, &fmul.operand1, &fmul.dest, BinOps::Mul)
+                }
+                Instruction::FDiv(fdiv) => {
+                    self.fl_bin_operations(&fdiv.operand0, &fdiv.operand1, &fdiv.dest, BinOps::Div)
+                }
                 _ => todo!(),
             }
         }
         match &bb.term {
-            Terminator::Ret(rec) => match rec.return_operand.as_ref().unwrap() {
-                LocalOperand { .. } => todo!(),
-                ConstantOperand(con_ref) => match con_ref.as_ref() {
-                    Constant::Int { bits: _, value } => {
-                        return Some(*value);
-                    }
-                    _ => todo!(),
+            Terminator::Ret(ret) => match ret.return_operand.as_ref() {
+                Some(op) => match op {
+                    LocalOperand { .. } => todo!(),
+                    ConstantOperand(con_ref) => match con_ref.as_ref() {
+                        Constant::Int { bits: _, value } => {
+                            return Some(*value);
+                        }
+                        _ => todo!(),
+                    },
+                    MetadataOperand => todo!(),
                 },
-                MetadataOperand => todo!(),
+                None => None,
             },
             _ => todo!(),
         }
     }
 
     fn store_var(&mut self, op: &Operand, val: &Operand) {
         match op {
-            LocalOperand { name, .. } => match val {
-                LocalOperand { .. } => todo!(),
-                ConstantOperand(con_ref) => match con_ref.as_ref() {
-                    Constant::Int { bits: _, value } => {
-                        self.mem.insert(name.clone(), *value);
-                    }
-                    _ => todo!(),
-                },
-                MetadataOperand => todo!(),
-            },
+            LocalOperand { name, .. } => {
+                let op_name = name;
+                match val {
+                    LocalOperand { name, .. } => self
```

Comment:
You could have just given this another name with `LocalOperand { name: newname, .. }`, but let's get this PR merged now. It's been long enough. :-)